### PR TITLE
[BarClerk-539] - [FE] Setup RTK for handling states and Redux Thunk for handling API request

### PR DESCRIPTION
**Pull request by impaulintech**
## Issue Link
- https://app.asana.com/0/1203234368773394/1203256158175539/f

## Definition of Done
- [ ] Redux was already setup

## Pre-condition
- yarn

## Expected Output
- Dev can request to the server

## Screenshots/Recordings

### CODE REVIEW

File: client/hooks/authMethods.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import toast from 'react-hot-toast';
+import { useRouter } from 'next/router';
+
+import { SignInUpFormValues } from 'shared/types';
+import { useAppDispatch, useAppSelector } from 'hooks/reduxSelector';
+import { reset, signIn, signOut, signUp } from 'redux/auth/authSlice';
+
+export const useAuthMethods = () => {
+  const { isError, error } = useAppSelector((state: any) => state.auth);
```
Comment (abduljalilpalala) [inline client/hooks/authMethods.ts:9]: State variable already has types.

```suggestion
  const { isError, error } = useAppSelector((state) => state.auth);
```

File: client/redux/auth/authService.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  return response.data;
+};
+
+const hydrateUserState = async (): Promise<any> => {
+  const response = await axios.get('/auth');
+  if (response.status === 200) {
+    return response.data;
+  }
+  return 'Something went wrong';
+};
```
Comment (abduljalilpalala) [inline client/redux/auth/authService.ts:44]: Errors will be thrown in the catch section in the thunk methods

```suggestion
const hydrateUserState = async (): Promise<any> => {
  const response = await axios.get('/auth');
  return response.data
};
```

File: client/redux/auth/authSlice.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  ActionReducerMapBuilder,
+} from '@reduxjs/toolkit';
+import { HYDRATE } from 'next-redux-wrapper';
+
+import authService from './authService';
+import { catchError } from 'utils/handleAxiosError';
+import { SignInUpFormValues, User, AxiosResponseError } from 'shared/types';
+
+type InitialState = {
+  user: any;
```
Comment (abduljalilpalala) [inline client/redux/auth/authSlice.ts:14]: Change `any` to `SignInUpFormValues`

File: client/redux/auth/authSlice.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  isError: boolean;
```
Comment (abduljalilpalala) [inline client/redux/auth/authSlice.ts:15]:
```suggestion
type InitialState = {
  user: SignInUpFormValues;
  isError: boolean;
```

File: client/shared/lib/axios.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import Axios from 'axios';
+import { getCookie } from 'cookies-next';
+
+const token = getCookie('token');
+
+export const axios: any = Axios.create({
+  baseURL: process.env.NEXT_PUBLIC_BACKEND_URL,
```
Comment (abduljalilpalala) [inline client/shared/lib/axios.ts:7]:
```suggestion

export const axios = Axios.create({
  baseURL: process.env.NEXT_PUBLIC_BACKEND_URL,
```

File: client/shared/types/index.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export type SignInUpFormFields =
+  | 'first_name'
+  | 'last_name'
+  | 'email'
+  | 'password'
+  | 'password_confirmation';
+
+export type User = any;
```
Comment (abduljalilpalala) [inline client/shared/types/index.ts:21]: Add the proper type instead of `any`.

**Review by abduljalilpalala — COMMENTED** [review]
I left some comments. Please check.

**Comment by johnpaul-sun** [comment]
Nicee, will work on this later. Thank you!!

**Comment by impaulintech** [comment]
All comments have been resolved!

File: client/redux/auth/authSlice.ts (hunk 4 above)
Comment (impaulintech) [reply client/redux/auth/authSlice.ts:15]: ahh for this one it should be like this

export type User = {
  user: {
    id: number;
    first_name: string;
    last_name: string;
    email: string;
    created_at: string;
    updated_at: string;
  };
  token: string;
};

export type InitialState = {
  user: User | null;
  isError: boolean;
  isSuccess: boolean;
  isLoading: boolean;
  error: AxiosResponseError;
};

**Pull request merged**